packages/gatsby-cli/src/reporter/redux/index.ts

```diff
@@ -46,6 +117,95 @@ export const dispatch = (action: ActionsUnion | Thunk): void => {
 
   store.dispatch(action)
 
+  // ignore diagnostic logs, otherwise diagnostic message itself will reset
```

Comment:
I understand what this block is doing. It seems to be quite verbose in terms of code for what it's doing (debouncing a "stuck" warning). Can it be abstracted a little bit into a global function? Closures may make that more difficult.

Reply:
I moved almost all diagnostics related code to its own file/module (other than calls to init and calling handler in our custom dispatch function). I didn't do much changes otherwise - while there are similarities in 2 timers that I add they are different enough that I'm not sure if there's much value in trying to DRY it too much)